bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/microsecend/TaggedValuesDeserializer.java

```diff
@@ -0,0 +1,47 @@
+package org.dataflowanalysis.analysis.converter.microsecend;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+public class TaggedValuesDeserializer extends JsonDeserializer<Map<String, List<String>>> {
+
+	@Override
+    public Map<String, List<String>> deserialize(JsonParser parser, DeserializationContext ctxt) throws IOException {
+        Map<String, List<String>> result = new HashMap<>();
+
+        while (parser.nextToken() != JsonToken.END_OBJECT) {
+            String fieldName = parser.getCurrentName();
+            parser.nextToken(); // Move to the value
+
+            if (parser.getCurrentToken() == JsonToken.START_ARRAY) {
+                // If it's an array, deserialize as List<String>
+                @SuppressWarnings("unchecked")
+				List<String> values = parser.readValueAs(List.class);
+                result.put(fieldName, values);
+            } else {
```

Comment:
Does this else case only catch single values during deserialization? We should probably throw an Exception here, when we find Token we do not expect